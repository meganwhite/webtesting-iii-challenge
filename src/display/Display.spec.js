// Test away!
import React from "react";
import renderer from "react-test-renderer"; // 1: install this npm module as a dev dependency
import { render, fireEvent } from "@testing-library/react";

import Display from './Display'

describe("<Display />", () => {
    it("matches snapshot", () => {
      const tree = renderer.create(<Display />); // generates a DOM tree
      expect(tree.toJSON()).toMatchSnapshot();
    });


    it('displays "Closed" if closed prop is true', () => {
        const { getByText } = render(<Display closed={true} />);
        expect(getByText('Closed'));
    })

    it('displays "Open" if closed prop is false', () => {
        const { getByText } = render(<Display closed={false} />);
        expect(getByText('Open'));
    })

    it('displays "Locked" if the locked prop is true', () => {
        const {getByText} = render(<Display locked={true} />);
        expect(getByText('Locked'));
    })

    it('displays "Unlocked" if the locked prop is true', () => {
        const {getByText} = render(<Display locked={false} />);
        expect(getByText('Unlocked'));
    })

  });
