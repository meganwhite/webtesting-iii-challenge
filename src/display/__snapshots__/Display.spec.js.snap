// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;

exports[`<Display /> matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
